### ISSUE TITLE
Accidentally deleting a container when actual wanted to delete a snapshot.

### ISSUE THREAD

**Issue by stoivo** [lxc/lxd#4747]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 16.04 xenial
 * The output of "lxc info" or if that fails:
```
config:
  storage.zfs_pool_name: lxd
api_extensions:
- id_map
- id_map_base
- resource_limits
api_status: stable
api_version: "1.0"
auth: trusted
auth_methods: []
public: false
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    ......................
    -----END CERTIFICATE-----
  certificate_fingerprint: 0272c243731fdec721dce38cd0a6f468448d7fc9c628b4e76cb0f13bfff28851
  driver: lxc
  driver_version: 2.0.8
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.4.0-128-generic
  server: lxd
  server_pid: 3264
  server_version: 2.0.11
  storage: zfs
  storage_version: "5"
```

# Issue description

Accidentally deleting a container when actual wanted to delete a snapshot.

# Steps to reproduce

 1. Running `lxc list` and seeing that you have a snapshot on one container which you want to remove
 1. Run `lxd show <container>` to find the snapshop name
 1. Accidentally run `lxd delete <container_name> <snapshot name>`, notice that I managed to forget `/`.
1. Get really mad on yourself for deleteing a container in production which is is daily usage but 10-15 employees.

# Information to attach

So this isn't really a bug. More of a request for making a bigger difference between deleting containers and deleting snapshots. Since one is very critical in production and the other is not. 
Another option is to have a confirmation when deleting a container. It's just so much work to recreate the container.

**Comment by laralar** [comment]
Agreed.. a little tricky..   it would be better to ask for confirmation, or have a different syntax for snapshots.  The same thing had happened to me.

**Comment by zertrin** [comment]
yeah I also got surprised by the default of not asking for confirmation for the destructive delete command.

Unfortunately, it will break backward compatibility with scripts using `lxc delete` but in interactive usage, prompting the user for confirmation by default unless an optional flag is set would be a good idea.

**Comment by stgraber** [comment]
We don't plan on changing the default behavior as this would indeed break scripts and various tools that wrap `lxc delete` and don't expect an interactive workflow.

One is to setup an alias from `lxc delete` to `lxc delete -i`:

```
stgraber@castiana:~$ lxc alias add delete "delete -i"
stgraber@castiana:~$ lxc delete c1
Remove c1 (yes/no): 
```

The other is to use the new deletion protection feature that got merged earlier today and will be available in LXD 3.3. With it in place, you can mark containers that should never be deleted with the `security.protection.delete=true` config key and any delete attempt from whatever means (CLI, API, ...) will be rejected until the key is unset.

**Comment by zertrin** [comment]
@laralar the problem is that **existing** scripts wouldn't know about it and would suddenly start to fail... This is a backward compatibility issue, so it's not possible to just change it like that.

**Comment by stoivo** [comment]
Is `lxc alias` supported for a long time? lxc 2?
Where is this this stored?

I wanted to raise the consern about accidely deleteing containers.

**Comment by laralar** [comment]
@zertrin this is completely understood. That is why I wrote the "LoL"

Sometimes keeping a design functionality because of past decisions is not the best choice though and it may be worth the effort of doing dinner manual work for a better design and usability.  I'm not saying that this is the case. That is for you to judge. But yes. I have deleted containers accidentally. I don't know how many people have done that or will do that in the future and the amount of effort required sometimes to rebuild a not backed up container may be more than changing a few scripts.

Maybe the alias can be created on the place where the scripts are run so that they don't have to be changed.

I'm not saying which designs better, keeping the delete without confirmation by default or changing it to a "-f required". I have my opinion. But I think that backwards compatibility is not always the best decision choice for keeping something how it is. 

Just my 2 cents.
